### ISSUE TITLE
Fix Knative NATSS Dispatcher bug

### ISSUE THREAD

**Issue by Abd4llA** [kyma-project/kyma#4810]
**Description**
NATSS dispatcher cannot send incoming events to a subscriber (ksvc) of a channel, if the label serving.knative.dev/visibility: cluster-local is set for the service.

Knative issue: [Eventing#1336](https://github.com/knative/eventing/issues/1336)

**Comment by marcobebway** [comment]
This issue is fixed in Knative 0.7.0. I have verified that subscribers can receive events in both cases with and without the ksvc label **serving.knative.dev/visibility: cluster-local**.

<details><summary>Click to see the knative service details</summary>
<p>

```yaml
$ kubectl get ksvc -n test-cluster-local my-service -oyaml

apiVersion: serving.knative.dev/v1beta1
kind: Service
metadata:
  labels:
    serving.knative.dev/visibility: cluster-local
  name: my-service
  namespace: test-cluster-local
  resourceVersion: "81547"
  selfLink: /apis/serving.knative.dev/v1beta1/namespaces/test-cluster-local/services/my-service
  uid: 31f455ae-b535-11e9-b144-42010a840121
spec:
  template:
    metadata:
      annotations:
        autoscaling.knative.dev/minScale: "1"
    spec:
      containers:
      - image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/message_dumper@sha256:ab5391755f11a5821e7263686564b3c3cd5348522f5b31509963afb269ddcd63
        name: user-container
        resources: {}
      timeoutSeconds: 300
  traffic:
  - latestRevision: true
    percent: 100
status:
  address:
    url: http://my-service.test-cluster-local.svc.cluster.local
  conditions:
  - lastTransitionTime: 2019-08-02T14:52:55Z
    status: "True"
    type: ConfigurationsReady
  - lastTransitionTime: 2019-08-02T14:52:56Z
    status: "True"
    type: Ready
  - lastTransitionTime: 2019-08-02T14:52:56Z
    status: "True"
    type: RoutesReady
  latestCreatedRevisionName: my-service-9mtzs
  latestReadyRevisionName: my-service-9mtzs
  observedGeneration: 1
  traffic:
  - latestRevision: true
    percent: 100
    revisionName: my-service-9mtzs
  url: http://my-service.test-cluster-local.svc.cluster.local
```

</p>
</details>


<details><summary>Click to see the subscriber logs</summary>
<p>

```yaml
$ kubectl logs -f -n test-cluster-local my-service-9mtzs-deployment-57fc4c9ccd-829k4 user-container

2019/08/02 14:53:13 Message Dumper received a message: POST / HTTP/1.1
Host: my-service.test-cluster-local.svc.cluster.local
Accept-Encoding: gzip
Ce-Beats: true
Ce-Heart: "yes"
Ce-Id: e8d44872-3803-4a9f-9949-f75086cad451
Ce-Knativehistory: "natss-kn-trigger-kn-channel.test-cluster-local.svc.cluster.local"
Ce-Source: https://github.com/knative/eventing/test/heartbeats/#test-cluster-local/heartbeats-sender-2klpl-69c9456566-wbxbq
Ce-Specversion: 0.2
Ce-The: 42
Ce-Time: 2019-08-02T14:53:13.579714151Z
Ce-Type: dev.knative.eventing.samples.heartbeat
Content-Length: 17
Content-Type: application/json
Forwarded: for=10.72.2.32;proto=http
User-Agent: Go-http-client/1.1
X-B3-Parentspanid: 386307661144b464
X-B3-Sampled: 0
X-B3-Spanid: 9bd146fa42e6bed6
X-B3-Traceid: 4f8d16ae06ee8e7cbcbddeb6fa492800
X-Envoy-Expected-Rq-Timeout-Ms: 600000
X-Envoy-Internal: true
X-Forwarded-For: 10.72.2.32, 127.0.0.1
X-Forwarded-Proto: http
X-Request-Id: 55b41011-0d43-9948-a04d-4bf4aca9858e
```

</p>
</details>

**Comment by marcobebway** [comment]
The issue https://github.com/knative/eventing/issues/1336 is closed.
